engines/buried/navarrow.cpp

```diff
@@ -0,0 +1,245 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * Additional copyright for this file:
+ * Copyright (C) 1995 Presto Studios, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "buried/buried.h"
+#include "buried/gameui.h"
+#include "buried/graphics.h"
+#include "buried/inventory_window.h"
+#include "buried/navarrow.h"
+#include "buried/navdata.h"
+#include "buried/resources.h"
+#include "buried/scene_view.h"
+
+#include "common/keyboard.h"
+#include "graphics/surface.h"
+
+namespace Buried {
+
+NavArrowWindow::NavArrowWindow(BuriedEngine *vm, Window *parent) : Window(vm, parent) {
+	_background = _vm->_gfx->getBitmap(IDB_ARROW_BACKGROUND);
+
+	_arrowBitmaps[0][0] = IDB_ARROW_UP_CLEAR;
+	_arrowBitmaps[0][1] = IDB_ARROW_UP_LIT;
+	_arrowBitmaps[0][2] = IDB_ARROW_UP_HIGHLIGHTED;
+	_arrowBitmaps[1][0] = IDB_ARROW_LEFT_CLEAR;
+	_arrowBitmaps[1][1] = IDB_ARROW_LEFT_LIT;
+	_arrowBitmaps[1][2] = IDB_ARROW_LEFT_HIGHLIGHTED;
+	_arrowBitmaps[2][0] = IDB_ARROW_RIGHT_CLEAR;
+	_arrowBitmaps[2][1] = IDB_ARROW_RIGHT_LIT;
+	_arrowBitmaps[2][2] = IDB_ARROW_RIGHT_HIGHLIGHTED;
+	_arrowBitmaps[3][0] = IDB_ARROW_DOWN_CLEAR;
+	_arrowBitmaps[3][1] = IDB_ARROW_DOWN_LIT;
+	_arrowBitmaps[3][2] = IDB_ARROW_DOWN_HIGHLIGHTED;
+	_arrowBitmaps[4][0] = IDB_ARROW_FORWARD_CLEAR;
+	_arrowBitmaps[4][1] = IDB_ARROW_FORWARD_LIT;
+	_arrowBitmaps[4][2] = IDB_ARROW_FORWARD_HIGHLIGHTED;
+
+	for (int i = 0; i < NUM_ARROWS; i++)
+		_arrowStatus[i] = BUTTON_DISABLED;
+
+	rebuildArrows();
+
+	_rect = Common::Rect(510, 292, 640, 418);
+}
+
+NavArrowWindow::~NavArrowWindow() {
+	_background->free();
+	delete _background;
+}
+
+bool NavArrowWindow::updateArrow(int button, int newStatus) {
+	_arrowStatus[button] = newStatus;
+
+	rebuildArrows();
+	invalidateWindow(false);
+	return true;
+}
+
+bool NavArrowWindow::updateAllArrows(int left, int up, int right, int down, int forward) {
+	// clone2727 says: This is wrong. Left and up are swapped.
+	// I can only imagine what bugs this causes.
+	_arrowStatus[0] = left;
```

Comment:
Another good candidate for putting a TODO prefix instead, since from the sounds of it, the problem was identified but not actually corrected.